### PR TITLE
Sean discussion

### PR DESCRIPTION
**Pull request by thecalliopecrashed**
@sjiribarren please comment on language translations.

### CODE REVIEW

File: src/languages/es.js [hunk 1]
```diff
@@ -333,7 +333,30 @@ export default {
...
-},
+  },
+
+  progress: {
+    title: `
+    Mi Progreso
+    `,
+
+    days: `
+    Days of Treatment (es)
```
Comment (thecalliopecrashed) [inline src/languages/es.js]:
```suggestion
    ~~~translation for the text~~~
```

File: db/migrate/20190311204446_add_fields_to_medication_reports.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class AddFieldsToMedicationReports < ActiveRecord::Migration[5.1]
+  def change
+    add_column :medication_reports, :took_medication, :boolean
+    add_column :medication_reports, :not_taking_medication_reason, :string
```
Comment (thecalliopecrashed) [inline db/migrate/20190311204446_add_fields_to_medication_reports.rb]: 👍 - I think we'll be able to re-use this field to store the multiple-choice result for #54.

File: db/migrate/20190311204446_add_fields_to_medication_reports.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class AddFieldsToMedicationReports < ActiveRecord::Migration[5.1]
+  def change
+    add_column :medication_reports, :took_medication, :boolean
```
Comment (thecalliopecrashed) [inline db/migrate/20190311204446_add_fields_to_medication_reports.rb]:
```suggestion
    add_column :medication_reports, :took_medication, :boolean, null: false
```

If you accept this suggestion, you'll have to:
```bash
docker-compose exec web rake db:rollback
docker-compose exec web rake db:migrate
```

Comment (thecalliopecrashed) [reply db/migrate/20190311204446_add_fields_to_medication_reports.rb]: After you run those, you should see the `null: false` designation in `db/schema.rb`.

File: public/index.html [hunk 4]
```diff
@@ -5,11 +5,14 @@
...
-    <link rel="manifest" href="/site.webmanifest">
-    <meta name="msapplication-TileColor" content="#00aba9">
+    <link rel="manifest" href="./images/site.webmanifest">
+    <link rel="apple-touch-icon" sizes="180x180" href="./images/apple-touch-icon.png">
+    <link rel="icon" type="image/png" sizes="32x32" href="./images/favicon-32x32.png">
+    <link rel="icon" type="image/png" sizes="16x16" href="./images/favicon-16x16.png">
+    <link rel="mask-icon" href="./images/safari-pinned-tab.svg" color="#5bbad5">
+    <link rel="shortcut icon" href="./images/favicon.ico">
+    <meta name="msapplication-TileColor" content="#9f00a7">
+    <meta name="msapplication-config" content="./images/browserconfig.xml">
```
Comment (thecalliopecrashed) [inline public/index.html]: Can you give a brief description of what changed here?
I don't recognize some of these tags.

File: src/Assembly.js [hunk 5]
```diff
@@ -274,11 +284,44 @@ class Assembly extends React.Component {
...
+          timestamp: `${this.survey_date}T${this.survey_medication_time}:00.000`,
+          took_medication: this.survey_tookMedication,
+          not_taking_medication_reason: this.survey_notTakingMedicationReason 
+        },
+        this.registration.information.uuid
+      )
+    }
+    
+    this.survey_date = null
+    this.survey_medication_time = null
```
Comment (thecalliopecrashed) [inline src/Assembly.js]: We need to set this back to the original value from the constructor, not `null`:

```suggestion
    this.survey_medication_time = DateTime.local().setLocale(this.locale).toLocaleString(DateTime.TIME_24_SIMPLE)
```

File: src/Assembly.js [hunk 6]
```diff
@@ -274,11 +284,44 @@ class Assembly extends React.Component {
...
+        { 
+          timestamp: `${this.survey_date}T${this.survey_medication_time}:00.000`,
+          took_medication: this.survey_tookMedication,
+          not_taking_medication_reason: this.survey_notTakingMedicationReason 
+        },
+        this.registration.information.uuid
+      )
+    }
+    
+    this.survey_date = null
```
Comment (thecalliopecrashed) [inline src/Assembly.js]: We need to set this back to the original value from the constructor, not `null`:

```suggestion
    this.survey_date = DateTime.local().setLocale(this.locale).toISODate()
```

File: src/Assembly.js [hunk 7]
```diff
@@ -274,11 +284,44 @@ class Assembly extends React.Component {
...
+    this.survey_tookMedication = null
+    this.survey_notTakingMedicationReason = null
   }
 
   reportSymptoms() {
-    this.registration.create("symptom_reports", this.symptoms, this.registration.information.uuid)
+    this.registration.create("symptom_reports", this.symptoms, 
+                            this.registration.information.uuid)
+
+    // Lousy way of reverting the survey back to normal
```
Comment (thecalliopecrashed) [inline src/Assembly.js]: Let's remove this comment – for now, this is the best option we have.

File: src/Assembly.js [hunk 8]
```diff
@@ -347,12 +390,17 @@ class Assembly extends React.Component {
...
-      </NavBar>
+      {/* Only shows the navbar when a user is logged in */}
+      <Observer>
+        {() => 
+          this.registration.information.uuid == null ? null : 
+            <NavBar>
+              <Navigation store={this} />
+            </NavBar>
+        }
+        </Observer>
```
Comment (thecalliopecrashed) [inline src/Assembly.js]: This looks good, but we'll likely want to apply the same logic to the `menu` element above.
Can you work some copy/paste magic?

File: src/components/MedicationReports.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+import { Icon } from "@mdi/react"
+import { mdiPill } from "@mdi/js"
+import Heading from "../primitives/Heading"
+import { Paragraph } from "reakit";
+
+import styled from "styled-components"
+
+const MedicationReports = observer(({ store }) => (
+      <div>
```
Comment (thecalliopecrashed) [inline src/components/MedicationReports.js]: This line (and the following) are indented too much - can you de-dent to two spaces?

File: src/components/MedicationReports.js [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+        <Fold>
+
+        <Question>
+        <Icon size="1.5rem" color={darkgrey} path={mdiPill} /> {store.translate("progress.medication")}
```
Comment (thecalliopecrashed) [inline src/components/MedicationReports.js]: Let's break up the component and the text onto each their own lines.
```suggestion
          <Icon size="1.5rem" color={darkgrey} path={mdiPill} />
          {store.translate("progress.medication")}
```

File: src/components/MedicationReports.js [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+        </Question>
+
+        { store.registration.information.medication_reports.length === 0
+          ? <Answer>
+              {/* TODO: Translate */}
+              <Info>No Medication Reports</Info>
```
Comment (thecalliopecrashed) [inline src/components/MedicationReports.js]:
```suggestion
              <Info>{store.translate("progress.no_medication_reports")}</Info>
```


`en.js` / `es.js`:
```json
medication_reports: "No Medication Reports",
```

File: src/components/MedicationReports.js [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+            </Answer>
+
+          : store.registration.information.medication_reports.map(({timestamp, id, took_medication,
+            not_taking_medication_reason}) => (
+            
+            // TODO: Sort by date and link reports
+            <Answer>
+              <Time key={id}>{DateTime.fromISO(timestamp).toLocaleString(DateTime.DATETIME_SHORT)}</Time>
+              <Info key={took_medication}>Took Medication: {took_medication && "Yes"} {!took_medication && "No, I reported \"" + not_taking_medication_reason + "\""}</Info>
```
Comment (thecalliopecrashed) [inline src/components/MedicationReports.js]:
```suggestion
              <Info key={took_medication}>
                {store.translate("progress.took_medication")}
                { took_medication
                ? store.translate("progress.took_medication_yes")
                : store.translate("progress.took_medication_no") + not_taking_medication_reason
                }
              </Info>
```

`en.js` / `es.js`:
```json
took_medication: "Took Medication:",
took_medication_yes: "Yes",
took_medication_no: "No; ",
```

File: src/components/Notes.js [hunk 13]
```diff
@@ -33,21 +34,21 @@ const Notes = observer(({ store }) => (
...
-    { store.notes.length === 0
+    { store.registration.information.notes.length === 0
       ? <Hint>{store.translate("notes.hint")}</Hint>
 
       : store.registration.information.notes.map(({ id, title, created_at, text, updated_at }) => (
         <Note key={id}>
           <Note.Header>
             <Note.Title>{title}</Note.Title>
-            <Note.Created>{created_at}</Note.Created>
+            <Note.Created>{DateTime.fromISO(created_at).toLocaleString(DateTime.DATETIME_SHORT)}</Note.Created>
```
Comment (thecalliopecrashed) [inline src/components/Notes.js]: Much better, thanks! I was about to open an issue for this.

File: src/components/Progress.js [hunk 14]
```diff
@@ -1,88 +1,47 @@
...
     <Provider theme={theme}>
-      <div>
-        {/* TODO: Translations for this whole page */}
-        <h2>Mi Progreso</h2>
+        <div>
+          <h2>{store.translate("progress.title")}</h2>
+        {/* TODO: Get rid of before merging */}
+        {/* {console.log("Medication Reports", patient_info.medication_reports)} */}
+        {/* {console.log("Symptom Reports", patient_info.symptom_reports)} */}
+        {/* {console.log("Information", store.registration.information.medication_reports)} */}
```
Comment (thecalliopecrashed) [inline src/components/Progress.js]: :x:

File: src/components/Progress.js [hunk 15]
```diff
@@ -1,88 +1,47 @@
...
 
         <ProgressBarContainer>
           <TreatmentProgress>
             {/* TODO: Link the data to this number */}
             <Percentage>{total_treatment_days}</Percentage>
 
-            {/* TODO translation */}
             <TreatmentDays>
-              <strong>Days of Treatment</strong>
+              <strong>{store.translate("progress.days")}</strong>
```
Comment (thecalliopecrashed) [inline src/components/Progress.js]: 👍

File: src/components/Progress.js [hunk 16]
```diff
@@ -1,88 +1,47 @@
...
 
-          <Answer>
-            {/* TODO: data */}
-            <P>2019/02/20 - Nausea</P>
-            <P>2019/02/19 - Redness, Nausea, Appetite loss</P>
-            <P>2019/02/18 - Lorem ipsum dolor sit amet consectetur adipisicing elit. Enim et alias nam obcaecati molestias expedita? Eos asperiores illum tempore, repudiandae, maiores molestiae natus aut accusantium voluptate quaerat eaque reiciendis quia?</P>
-            <P>2019/02/17 - Null</P>
-          </Answer>
-        </Fold>
+        <MedicationReports store={store}></MedicationReports>
```
Comment (thecalliopecrashed) [inline src/components/Progress.js]:
```suggestion
        <MedicationReports store={store} />
```

File: src/components/Progress.js [hunk 17]
```diff
@@ -1,88 +1,47 @@
...
+        <SideEffects store={store}></SideEffects>
```
Comment (thecalliopecrashed) [inline src/components/Progress.js]:
```suggestion
        <SideEffects store={store} />
```

File: src/components/Progress.js [hunk 18]
```diff
@@ -1,88 +1,47 @@
...
+        <StripReports store={store}> </StripReports>
```
Comment (thecalliopecrashed) [inline src/components/Progress.js]:
```suggestion
        <StripReports store={store} />
```

File: src/components/ReportMedication.js [hunk 19]
```diff
@@ -38,17 +38,16 @@ const ReportMedication = observer(({ store }) => (
           <TextFieldLabel>
             <span>{store.translate("survey.tookMedication.reason")}</span>
 
-            {/* TODO: Make this TextInput expandable,
-            or give a list of reasons to the patient */}
-
-            <TextInput use="textarea" />
+            <TextInput use="textarea"
```
Comment (thecalliopecrashed) [inline src/components/ReportMedication.js]:
```suggestion
            <TextInput
              use="textarea" 
```

Multiple lines!

File: src/components/SideEffects.js [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+
+import { Icon } from "@mdi/react"
+import { mdiFormatListChecks } from "@mdi/js"
+import Heading from "../primitives/Heading"
+import { Paragraph } from "reakit";
+
+import styled from "styled-components"
+
+const SideEffects = observer(({ store }) => (
+      <div>
```
Comment (thecalliopecrashed) [inline src/components/SideEffects.js]: De-indent?

File: src/components/SideEffects.js [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+        <Fold>
+          <Question>
+          <Icon size="1.5rem" color={darkgrey} path={mdiFormatListChecks} /> {store.translate("progress.side_effect")}
```
Comment (thecalliopecrashed) [inline src/components/SideEffects.js]:
```suggestion
            <Icon size="1.5rem" color={darkgrey} path={mdiFormatListChecks} />
            {store.translate("progress.side_effect")}
```

File: src/components/SideEffects.js [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+          </Question>
+
+          { store.registration.information.symptom_reports.length === 0
+            ? <Answer>
+                {/* TODO: Translate this piece */}
+                <Info>No Side Effects Reported</Info>
```
Comment (thecalliopecrashed) [inline src/components/SideEffects.js]:
```suggestion
                <Info>{store.translate("progress.no_side_effects")}</Info>
```

`en.js` / `es.js`:
```json
no_side_effects: "No Side Effects Reported",
```

File: src/components/SideEffects.js [hunk 23]
```diff
@@ -0,0 +1,64 @@
...
+              
+              // TODO: Sort by date and link reports
+              <Answer>
+                <Time key={created_at}>{DateTime
+                .fromISO(created_at)
+                .toLocaleString(DateTime.DATETIME_SHORT)}</Time>
+                {reported_symptoms.length !== 0 ? 
+                <Info key={reported_symptoms}>{reported_symptoms.join(", ")}</Info>
+                  // TODO: Translations
+                : <Info>N/A</Info>}
```
Comment (thecalliopecrashed) [inline src/components/SideEffects.js]:
```suggestion
                : <Info>{store.translate("progress.no_side_effects")}</Info>
                }
```

This translation is already supplied by the previous comment; we can use the same wording in both places.

File: src/components/StripReports.js [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+import Heading from "../primitives/Heading"
+import { Image } from "reakit";
+
+import styled from "styled-components"
+
+const StripReports = observer(({ store }) => (
+      <div>
+        <Fold>
+          <Question>
+          <Icon size="1.5rem" color={darkgrey} path={mdiCamera} /> {store.translate("progress.test_result")}
```
Comment (thecalliopecrashed) [inline src/components/StripReports.js]:
```suggestion
            <Icon size="1.5rem" color={darkgrey} path={mdiCamera} />
            {store.translate("progress.test_result")}
```

File: src/components/StripReports.js [hunk 25]
```diff
@@ -0,0 +1,64 @@
...
+          </Question>
+
+          { store.registration.information.strip_reports.length === 0
+            ? <Answer>
+                {/* TODO: Need translation */}
+                <Info>No Strip Reports Submitted</Info>
```
Comment (thecalliopecrashed) [inline src/components/StripReports.js]:
```suggestion
                <Info>{store.translate("progress.no_strip_reports")}</Info>
```

`en.js` / `es.js`:
```js
no_strip_reports: "No Strip Reports Submitted",
```

File: src/components/StripReports.js [hunk 26]
```diff
@@ -0,0 +1,64 @@
...
+const StripReports = observer(({ store }) => (
+      <div>
+        <Fold>
+          <Question>
+          <Icon size="1.5rem" color={darkgrey} path={mdiCamera} /> {store.translate("progress.test_result")}
+          </Question>
+
+          { store.registration.information.strip_reports.length === 0
+            ? <Answer>
+                {/* TODO: Need translation */}
```
Comment (thecalliopecrashed) [inline src/components/StripReports.js]:
```suggestion
```

File: src/components/StripReports.js [hunk 27]
```diff
@@ -0,0 +1,64 @@
...
+                <Info>No Strip Reports Submitted</Info>
+              </Answer>
+
+            : store.registration.information.strip_reports.map(({created_at, photo, status}) => (
+              <Answer>
+                <Time key={created_at}>{DateTime
```
Comment (thecalliopecrashed) [inline src/components/StripReports.js]:
```suggestion
                <Time key={created_at}>
                  {DateTime
```

File: src/components/StripReports.js [hunk 28]
```diff
@@ -0,0 +1,64 @@
...
+                  .fromISO(created_at)
+                  .toLocaleString(DateTime.DATETIME_SHORT)}
+                </Time>
+                <Info>
+                  {/* TODO: Need translations */}
+                  {status || "Not Reviewed Yet"}
```
Comment (thecalliopecrashed) [inline src/components/StripReports.js]:
```suggestion
                  {status || store.translate("progress.not_reviewed")}
```

`en.js` / `es.js`:
```js
not_reviewed: "Not Reviewed Yet",
```

File: public/index.html (hunk 4 above)
Comment (campbellsean) [reply public/index.html]: I used https://realfavicongenerator.net/#.XHYEmpNKhTY for generating the Favicon and related app designs. The ones referenced above are for Apple's Touchbar and Microsofts Tile design when you click the window button on Windows.

File: src/components/MedicationReports.js (hunk 12 above)
Comment (campbellsean) [reply src/components/MedicationReports.js]: Clean

File: src/components/SideEffects.js (hunk 23 above)
Comment (campbellsean) [reply src/components/SideEffects.js]: I think it is a little lengthly but lets keep it for now

File: src/Assembly.js (hunk 8 above)
Comment (campbellsean) [reply src/Assembly.js]: Good catch

**Pull request merged**